streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -346,24 +348,29 @@ public void writeAsText(final String filePath,
         return branchChildren;
     }
 
-    public static <K, V> KStream<K, V> merge(final InternalStreamsBuilder builder,
-                                             final KStream<K, V>[] streams) {
-        if (streams == null || streams.length == 0) {
-            throw new IllegalArgumentException("Parameter <streams> must not be null or has length zero");
-        }
+    
+    @Override 
+    public KStream<K, V> merge(final KStream<K, V> stream) {
+        return merge(builder, stream);
+    }
+    
+    private KStream<K, V> merge(final InternalStreamsBuilder builder,
+                               final KStream<K, V> stream) {
 
         String name = builder.newName(MERGE_NAME);
-        String[] parentNames = new String[streams.length];
+        String[] parentNames = new String[2];
         Set<String> allSourceNodes = new HashSet<>();
         boolean requireRepartitioning = false;
-
-        for (int i = 0; i < streams.length; i++) {
-            KStreamImpl<K, V> stream = (KStreamImpl<K, V>) streams[i];
-
-            parentNames[i] = stream.name;
-            requireRepartitioning |= stream.repartitionRequired;
-            allSourceNodes.addAll(stream.sourceNodes);
-        }
+        
+        parentNames[0] = this.name;
+        requireRepartitioning |= repartitionRequired;
+        allSourceNodes.addAll(sourceNodes);
+        
+        KStreamImpl<K, V> streamCast1 = (KStreamImpl<K, V>) stream;
```

Comment:
nit: we usually do not use unnecessary numbers as part of the parameter; rename to `streamImpl` instead.